alas_but_one.py

```diff
@@ -35,13 +71,16 @@
                 print(e)
 
 ctr = Counter(word_list)
-if os.path.isfile(EXCLUSION_FILE):
-    exclusion_list = open(EXCLUSION_FILE).read()
-    for w,v in sorted(ctr.items(), key=lambda pair: pair[1], reverse=True):
-        if (v <= MAX_OCCUR):
-            if (not((w + " ") in exclusion_list)):
-                print("%s [%d]" % (w,v), file=open(OUTPUT_FILE, 'a'))
-else:
-    for w,v in sorted(ctr.items(), key=lambda pair: pair[1], reverse=True):
-        if (v <= MAX_OCCUR):
-            print("%s [%d]" % (w,v), file=open(OUTPUT_FILE, 'a'))
+for w,v in sorted(ctr.items(), key=lambda pair: pair[1], reverse=True):
+    addWord=((v <= MAX_OCCUR) and
+             (not(doExclusion) or
+                (doExclusion and not(w in exclusion_list))))
+    if addWord:
+        if args.noCount:
+            outputString="%s" % (w)
+        else:
+            outputString="%s [%d]" % (w,v)
+        if args.output:
+            print(outputString, file=open(args.output, 'a'))
```

Comment:
Issue:

I think that just because ``file`` loses scope after the ``print`` call doesn't mean that the file gets closed. Even if closed, opening a file for every word to be printed is much less efficient than opening the file once and keeping a reference to the file handle.

Suggestion:

A few different paths:
1) Skip the output file as it offers the exact level of functionality as piping stdout to a file (e.g. ``<script + args> > output.txt``).

2) Collect all the words for the output before opening the file to write to such that they can all be written at once.
Use the ``with open(output_file, "w") as output_filehandle:`` statement to avoid having to explicitly call close on the file handle. Omit the truncate(0) call.

3) It might generate more rounds of review, but if there was some planned advantage to using the output option that didn't make it to this revision, implement it.